packages/lodestar/src/network/interface.ts

```diff
@@ -10,16 +10,19 @@ import {
   BeaconBlocksByRootResponse,
   Goodbye,
   RequestBody,
-  ResponseBody, Status,
+  ResponseBody, Status, SignedBeaconBlock,
 } from "@chainsafe/eth2.0-types";
 
 import {Method, RequestId} from "../constants";
 import StrictEventEmitter from "strict-event-emitter-types";
 import {IGossip} from "./gossip/interface";
 
 // req/resp
+export type Response = {err?: Error; output?: ResponseBody};
 
-export type ResponseCallbackFn = (err: Error|null, output: ResponseBody|null) => void;
+export type ResponseCallbackFn = (response: Response) => void;
+
+export type ChunkResponse = {err?: Error; output: Status | Goodbye | SignedBeaconBlock};
```

Comment:
Can you call this `ResponseChunk`?
We shouldn't need a `Response` and a `ResponseChunk` type. (every response should be a `ResponseChunk`)
If we do have a higher level `Response`-ish interface it should expose an `Iterable<ResponseChunk>` rather than an array, eg: `SignedBeaconBlock[]`

See the "relaxed BNF grammer" code section in the spec:
https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/p2p-interface.md#reqresp-interaction